test/powershell/Modules/Microsoft.PowerShell.Core/Out-Default.Tests.ps1

```diff
@@ -0,0 +1,40 @@
+Describe "Out-Default Tests" -tag CI {
+    BeforeAll {
+        # due to https://github.com/PowerShell/PowerShell/issues/3405, `Out-Default -Transcript` emits output to pipeline
+        # as running in Pester effectively wraps everything in parenthesis, workaround is to use another powershell
+        # to run the test script passed as a string
+        $powershell = "$PSHOME/powershell"
+    }
+
+    It "'Out-Default -Transcript' shows up in transcript, but not host" {
+        $script = @"
+            `$null = Start-Transcript -Path "$testdrive\transcript.txt";
+            'hello' | Microsoft.PowerShell.Core\Out-Default -Transcript;
+            'bye';
+            `$null = Stop-Transcript
+"@
+
+        & $powershell -c $script | Should BeExactly 'bye'
+        "TestDrive:\transcript.txt" | Should Contain 'hello'
+    }
+
+    It "Out-Default reverts transcription state when used more than once in a pipeline" {
+        & $powershell -c "Out-Default -Transcript | Out-Default -Transcript; 'Hello'" | Should BeExactly "Hello"
+    }
+
+    It "Out-Default reverts transcription state when exception occurs in pipeline" {
+        & $powershell -c "try { & { throw } | Out-Default -Transcript } catch {}; 'Hello'" | Should BeExactly "Hello"
+    }
+
+    It "Out-Default reverts transcription state even if Dispose() isn't called" {
```

Comment:
>Isn't this just a programmer error to not Dispose?

But, how to reliable call `Dispose` from PowerShell? Assume I wrap `Out-Default` in `SteppablePipeline` to make a proxy command with customized behavior:

```powershell
New-Item Function:Out-Default -Value ([System.Management.Automation.ProxyCommand]::Create((Get-Command Out-Default)))
```
Then someone throw exception in pipeline with `Out-Default -Transcript`:
```powershell
& {throw} | Out-Default -Transcript
```
And `Dispose` of proxy command is not called, because PowerShell functions does not have a `Dispose`. As result `Dispose` of `SteppablePipeline` is not called, which lead to `Dispose` of wrapped `Out-Default` not called as well.